packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx

```diff
@@ -83,6 +83,12 @@ export const ConfirmModal = ({
     }
   }, [isOpen, confirmationText]);
 
+  const onConfirmClick = async () => {
+    setDisabled(true);
+    await onConfirm();
+    setDisabled(false);
```

Comment:
Hmm. I think we should wrap this in try/catch, or just use promise syntax with .finally()


```ts
const onConfirmClick = async () => {
  const value = onConfirm();

  if (!value || !('finally' in value)) {
    return;
  }

  // value is a promise
  setDisabled(true);
  value.finally(() => setDisabled(false));
};
```

Otherwise if onConfirm throws/rejects the button will be stuck in disabled state.